jax/_src/numpy/lax_numpy.py

```diff
@@ -1525,6 +1525,27 @@ def union1d(ar1, ar2):
   return unique(conc)
 
 
+@_wraps(np.setxor1d)
+def setxor1d(ar1, ar2, assume_unique=False):
+  ar1 = core.concrete_or_error(asarray, ar1, "The error arose in setxor1d()")
+  ar2 = core.concrete_or_error(asarray, ar2, "The error arose in setxor1d()")
+
+  if ar1.ndim != 1 or ar2.ndim != 1:
+    raise ValueError("only 1-dimensional input supported.")
```

Comment:
It looks like numpy's `setxor1d` will ravel inputs:
```python
>>> import numpy as np
>>> np.setxor1d(np.ones((2, 2)), np.zeros((2, 2)))
array([0., 1.])
```
I think we should match those semantics here.

Reply:
One thing to note is that `np.setxor1d` does not explicitly ravel the inputs, it just happens as a by-product from call to `np.unique`. 
```python
>>> import numpy as np
>>> np.setxor1d(np.ones((2, 2)), np.zeros((2, 2)))
array([0., 1.])

>>> np.setxor1d(np.ones((2, 2)), np.zeros((2, 2)), assume_unique=True)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-4-1b7ce8d6b603> in <module>
----> 1 np.setxor1d(np.ones((2, 2)), np.zeros((2, 2)), assume_unique=True)

<__array_function__ internals> in setxor1d(*args, **kwargs)

~/miniconda3/envs/jaxenv/lib/python3.8/site-packages/numpy/lib/arraysetops.py in setxor1d(ar1, ar2, assume_unique)
    485 
    486     aux.sort()
--> 487     flag = np.concatenate(([True], aux[1:] != aux[:-1], [True]))
    488     return aux[flag[1:] & flag[:-1]]
    489 

<__array_function__ internals> in concatenate(*args, **kwargs)

ValueError: all the input arrays must have same number of dimensions, but the array at index 0 has 1 dimension(s) and the array at index 1 has 2 dimension(s)
```

So should we replicate this as well or explicitly flatten the input arrays?

Reply:
I'd lean toward explicitly flattening the input arrays: this matches what is done by `np.setdiff1d`, `np.union1d` and `np.intersect1d`.